### ISSUE TITLE
backward sync not working if buffer is modified

### ISSUE THREAD

**Issue by clason** [lervag/vimtex#924]
As the title says: backward sync from PDF to source is not working if there are unsaved changes in the buffer; this is with neovim (HEAD), neovim-remote (1.7.4, but previous versions had the same behaviour), both on Linux (okular) and macOS (skim). 

### Minimal working example

Any.

### Minimal vimrc file

for Linux: 

```vim
set nocompatible

" Load Vimtex
let &rtp  = '~/.vim/bundle/vimtex,' . &rtp
let &rtp .= ',~/.vim/bundle/vimtex/after'

" Load other plugins, if necessary
" let &rtp = '~/path/to/other/plugin,' . &rtp

filetype plugin indent on
syntax enable

" Vimtex options go here
let g:vimtex_compiler_progname = 'nvr'
let g:vimtex_view_general_viewer = 'okular'
let g:vimtex_view_general_options = '--unique file:@pdf\#src:@line@tex'
let g:vimtex_view_general_options_latexmk = '--unique'
```

For macOS, replace the last three lines with

```
let g:vimtex_view_method = 'skim'
```

**Comment by lervag** [comment]
Strange. This seems to work well here. I'm uploading the test files I've used. I've done `nvim -u minivimrc`, then started compilation with `\ll`, then `gg=G`, then moved to one of the last lines and done `\lv`. It seems to work as expected.

**Comment by clason** [comment]
I might be confused here, but isn't this *forward* sync? Backward sync is from PDF to TeX, i.e., clicking somewhere in okular or skim and jumping to the corresponding line in (neo)vim.

**Comment by lervag** [comment]
Yes, you're right. Let me try again...

**Comment by lervag** [comment]
Ok, I've reproduced. This also happens with `zathura`, and so I've changed the test files to use zathura. It does not happen with Vim, though, and I have no idea why this would occur. As far as I know, there has been no change to vimtex that could have caused this.

I've now tested this with `neovim 0.1.6`, as well as `neovim-remote 1.5.2`. It seems this might have never really worked, but I am not quite sure. I'll test some older versions of vimtex as well.

**Comment by clason** [comment]
Then this might actually be by design (after all, neovim removed the automatic reload of changed files...) Thanks for looking into this in so much detail! Is there any way of checking whether it's a neovim or neovim-remote issue?

**Comment by lervag** [comment]
I've also tried with a two year old version of vimtex, and I observe the same behaviour. If I issue the remote command from the terminal, I get the following error:

```
Traceback (most recent call last):
  File "/usr/bin/nvr", line 11, in <module>
    load_entry_point('neovim-remote==1.7.2', 'console_scripts', 'nvr')()
  File "/usr/lib/python3.6/site-packages/nvr/nvr.py", line 334, in main
    neovim.execute(options.remote_silent + arguments, 'edit', silent=True)
  File "/usr/lib/python3.6/site-packages/nvr/nvr.py", line 106, in execute
    self.server.command('{} {}'.format(cmd, prepare_filename(fname)))
  File "/usr/lib/python3.6/site-packages/neovim/api/nvim.py", line 218, in command
    return self.request('nvim_command', string, **kwargs)
  File "/usr/lib/python3.6/site-packages/neovim/api/nvim.py", line 131, in request
    res = self._session.request(name, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/neovim/msgpack_rpc/session.py", line 98, in request
    raise self.error_wrapper(err)
neovim.api.nvim.NvimError: b'Vim(edit):E37: No write since last change (add ! to override)'
```

I'm trying to use other calls instead now, instead of the `--remote` (or `--remote-silent`).

**Comment by clason** [comment]
Good call! (Why didn't I think of this?) That does look vaguely familiar -- I thought it might be something like https://github.com/autozimu/LanguageClient-neovim/issues/5, but `set hidden` doesn't seem to help. Maybe you have to add the `-s` option to the `nvr` call somewhere?

(So it's definitely not a vimtex issue -- except possibly regarding documentation.)

**Comment by clason** [comment]
Since it's not a vimtex issue, should I report it at https://github.com/mhinz/neovim-remote?

**Comment by lervag** [comment]
Yes, I think that sounds like a good idea. However, I would try to make the issue report independent. I would guess that you could make a minimal vimrc file to start, then open neovim on a simple test file, make some change, and finally call `nvr` from the command line to see the error. I guess it will also be educational to do the `nvr` call first before you make a change to the file to show that it works as expected.

**Comment by lervag** [comment]
If you do open such an issue, please link it here. I would be interested to follow the issue thread.

**Comment by clason** [comment]
That was exactly my plan!

**Comment by lervag** [comment]
Hehe, great minds think alike (or so they say) . :)

**Comment by lervag** [comment]
Wow, that's fast. Great!
